CMakeLists.txt

```diff
@@ -1,4 +1,10 @@
 
+# Disable in-source builds to prevent source tree corruption.
+if( "${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}" )
+  message( FATAL_ERROR "FATAL: In-source builds are not allowed.
+                       You should create a separate directory for build files." )
+endif()
```

Comment:
I actually don't care about case anymore, but in the past we had wanted to follow Trilinos use of upper case. @dalg24

Reply:
I can't find any reference to **Trilinos** cmake coding style. Could you explain a little further?